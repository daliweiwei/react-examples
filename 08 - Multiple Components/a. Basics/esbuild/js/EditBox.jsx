/*! European Union Public License version 1.2 !*/
/*! Copyright Â© 2018-2024 Rick Beerendonk     !*/

function EditBox({ onChange, value }) {
  function handleChange(e) {
    onChange(e.target.value);
  }

  return <input onChange={handleChange} value={value} />;
}

export default EditBox;
