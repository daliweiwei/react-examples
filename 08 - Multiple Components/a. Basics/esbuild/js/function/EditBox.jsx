/*! European Union Public License version 1.2 !*/
/*! Copyright Â© 2018 Rick Beerendonk          !*/

/* Inline function */

function EditBox({ onChange, value }) {
  return <input onChange={e => onChange(e.target.value)} value={value} />;
}

/* Function */

/*
function EditBox({ onChange, value }) {
  function handleChange(e) {
    return onChange(e.target.value);
  }

  return (
    <input onChange={handleChange} value={value} />
  );
}
*/

/* Arrow */

/*
function EditBox({ onChange, value }) {
  const handleChange = e => onChange(e.target.value);
  return (
    <input onChange={handleChange} value={value} />
  );
}
*/

export default EditBox;
