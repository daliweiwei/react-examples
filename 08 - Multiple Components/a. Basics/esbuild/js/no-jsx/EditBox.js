/*! European Union Public License version 1.2 !*/
/*! Copyright Â© 2018 Rick Beerendonk          !*/

import { createElement } from 'react';

function EditBox({ onChange, value }) {
  return createElement('input', {
    autoFocus: true,
    onChange: e => onChange(e.target.value),
    value: value
  });
}

export default EditBox;
