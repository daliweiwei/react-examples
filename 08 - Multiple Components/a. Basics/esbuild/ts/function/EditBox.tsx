/*! European Union Public License version 1.2 !*/
/*! Copyright Â© 2018 Rick Beerendonk          !*/

interface EditBoxProps {
  onChange: (value: string) => void;
  value: string;
}

/* Inline function */

function EditBox({ onChange, value }: EditBoxProps) {
  return (
    <input autoFocus onChange={e => onChange(e.target.value)} value={value} />
  );
}

/* Separate function */

/*
function EditBox({ onChange, value }: EditBoxProps) {
  function handleChange(e: ChangeEvent<HTMLInputElement>) {
    onChange(e.target.value);
  }

  return (
    <input autoFocus onChange={handleChange} value={value} />
  );
}
*/

export default EditBox;
