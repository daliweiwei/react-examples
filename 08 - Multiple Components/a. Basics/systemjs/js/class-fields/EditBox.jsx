/*! European Union Public License version 1.2 !*/
/*! Copyright Â© 2018 Rick Beerendonk          !*/

import React from 'react';

function EditBox({ onChange, value }) {
  return (
    <input autoFocus onChange={e => onChange(e.target.value)} value={value} />
  );
}

export default EditBox;
