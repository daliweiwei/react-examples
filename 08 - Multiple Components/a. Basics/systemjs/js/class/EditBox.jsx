/*! European Union Public License version 1.2 !*/
/*! Copyright Â© 2018 Rick Beerendonk          !*/

import React from 'react';

/* Inline function */

function EditBox({ onChange, value }) {
  return (
    <input autoFocus onChange={e => onChange(e.target.value)} value={value} />
  );
}

/* method */

/*
class EditBox extends React.Component {
  constructor(props) {
    super(props);

    this.handleChange = this.handleChange.bind(this);
  }
  handleChange(e) {
    this.props.onChange(e.target.value);
  }
  render() {
    return (
      <input autoFocus onChange={this.handleChange} value={this.props.value} />
    );
  }
}
*/

export default EditBox;
