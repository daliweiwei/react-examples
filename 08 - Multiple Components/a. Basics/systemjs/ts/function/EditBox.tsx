/*! European Union Public License version 1.2 !*/
/*! Copyright Â© 2018 Rick Beerendonk          !*/

import React from 'react';

/* Inline function */

interface IEditBoxProps {
  onChange: (value: string) => void;
  value: string;
}

const EditBox: React.FC<IEditBoxProps> = ({ onChange, value }) => (
  <input autoFocus onChange={e => onChange(e.target.value)} value={value} />
);

/* Arrow */

/*
const EditBox: React.FC<IEditBoxProps> = ({ onChange, value }) {
  const handleChange = e => onChange(e.target.value);
  return (
    <input autoFocus onChange={e => onChange(e.target.value)} value={value} />
  );
}
*/

export default EditBox;
