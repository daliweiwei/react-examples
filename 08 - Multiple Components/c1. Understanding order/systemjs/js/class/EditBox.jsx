/*! European Union Public License version 1.2 !*/
/*! Copyright Â© 2018 Rick Beerendonk          !*/

/* eslint no-console:"off" */

import React from 'react';

class EditBox extends React.Component {
  constructor(props) {
    /***/ console.log('EditBox - constructor()');

    super(props);

    this.handleChange = this.handleChange.bind(this);
  }
  handleChange(e) {
    /***/ console.log('EditBox - handleChange()');

    this.props.onChange(e.target.value);
  }
  render() {
    /***/ console.log('EditBox - render()');

    return (
      <input autoFocus onChange={this.handleChange} value={this.props.value} />
    );
  }
}

export default EditBox;
