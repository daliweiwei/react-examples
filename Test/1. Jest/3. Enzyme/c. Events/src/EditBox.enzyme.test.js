import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import EditBox from './EditBox';

configure({ adapter: new Adapter() });

/// Enzyme ///

describe('EditBox', () => {
  /// -- Property -- ///

  it('Does not throw when no name is provided (enzyme)', () => {
    const wrapper = shallow(<EditBox />);
    expect(wrapper.html()).toEqual('<input value=""/>');
  });

  it('renders name when provided (enzyme)', () => {
    const wrapper = shallow(<EditBox value="World" />);

    expect(wrapper.type()).toEqual('input');

    // Use the property...
    expect(wrapper.at(0).props().value).toEqual('World');

    // ...or the whole node
    expect(wrapper.html()).toEqual('<input value="World"/>');
  });

  it('renders world snapshot when name is world (enzyme)', () => {
    const wrapper = shallow(<EditBox value="World" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  /// -- Event -- ///

  it('raises onChange event when changed (enzyme)', () => {
    const handleChange = jest.fn();
    const wrapper = shallow(<EditBox onChange={handleChange} name="World" />);

    wrapper.find('input').simulate('change', { target: { value: 'React' } });

    expect(handleChange).toHaveBeenCalledTimes(1);
    expect(handleChange).toHaveBeenCalledWith('React');
  });
});
