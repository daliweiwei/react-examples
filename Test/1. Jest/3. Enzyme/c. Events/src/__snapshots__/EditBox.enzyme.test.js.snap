// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditBox renders world snapshot when name is world (enzyme) 1`] = `
<input
  onChange={[Function]}
  value="World"
/>
`;
